File: trunk/Examples.VB/ATM/Domain/Account.vb
Public Class Account
    Implements IAccount

    Private _balance As Integer = 0

    Public Sub Transfer(ByVal amount As Integer, ByVal toAccount As IAccount) Implements IAccount.Transfer
        _balance -= amount
        toAccount.Balance += amount
    End Sub


    Public Property Balance() As Integer Implements IAccount.Balance
        Get
            Return _balance
        End Get
        Set(ByVal value As Integer)
            _balance = value
        End Set
    End Property

End Class

File: trunk/Examples.CS/ATM/Domain/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Examples.CS.ATM.Domain
{
    public class Account : IAccount
    {
        private int _balance = 0;

        public void Transfer(int amount, IAccount toAccount)
        {
            _balance -= amount;
            toAccount.Balance += amount;
        }


        public int Balance
        {
            get { return _balance; }
            set { _balance = value; }
        }


    }
}
